### PR TITLE
Fix getresultvalues

### PR DESCRIPTION
**Pull request by lsetiawan**
## Overview

This PR addresses #127. It changes the lowercase dataframe columns to the same cases as the model colump properties.

### CODE REVIEW

**Comment by emiliom** [comment]
@lsetiawan I have a broad question: this is a PR on master. But there is a bunch of commits on a dev branch that @Elijahwalkerwest will merge into master soon, hopefully early next week, right?

Given that, is it better to wait until Eli has merged his accumulated commits into master? Or are you ok with merging this PR now and possibly needing to spend time resolving conflicts later?

File: odm2api/ODM2/services/readService.py [hunk 1]
```diff
@@ -120,10 +120,23 @@ def assignRelatedFeatures(self, relatedfeatures):
                     self.related_features = related
 
 
+class ReadODM2(serviceBase):
+    def _get_columns(self, model):
```
Comment (emiliom) [inline odm2api/ODM2/services/readService.py]: Nice! :+1:

File: odm2api/ODM2/services/readService.py [hunk 2]
```diff
@@ -1261,39 +1274,47 @@ def getResultDerivationEquations(self):
...
+            >>> READ = ReadODM2(SESSION_FACTORY)
+            >>> READ.getResultValues(resultids=[10, 11])
+            >>> READ.getResultValues(resultids=[100, 20, 34], starttime=datetime.today())
+            >>> READ.getResultValues(resultids=[1, 2, 3, 4],
+            >>>     starttime=datetime(2000, 01, 01),
+            >>>     endtime=datetime(2003, 02, 01))
 
         """
-        type = self._session.query(Results).filter_by(ResultID=resultids[0]).first().ResultTypeCV
+        restype = self._session.query(Results).filter_by(ResultID=resultids[0]).first().ResultTypeCV
```
Comment (emiliom) [inline odm2api/ODM2/services/readService.py]: Thanks for using this opportunity to eliminate the use of reserved words (`type`), specially in a case like this one where it'll have no impact at all to code using this function.

Comment (emiliom) [reply odm2api/ODM2/services/readService.py]: Actually, come to think of it, there's already a variable below called `ResultType`. `restype` and `ResultType` seem too similar, and that's confusing. How about renaming `ResultType` to `ResultValues`, assuming that doesn't lead to other conflicts or confusion?

File: odm2api/ODM2/services/readService.py [hunk 3]
```diff
@@ -1310,6 +1331,7 @@ def getResultValues(self, resultids, starttime=None, endtime=None):
                 con=self._session_factory.engine,
                 params=query.params
             )
+            df.columns = [self._get_columns(ResultType)[c] for c in df.columns]
```
Comment (emiliom) [inline odm2api/ODM2/services/readService.py]: :+1:

**Comment by lsetiawan** [comment]
> @lsetiawan I have a broad question: this is a PR on master. But there is a bunch of commits on a dev branch that @Elijahwalkerwest will merge into master soon, hopefully early next week, right?

This is a PR to the dev branch that we have been merging into. This dev branch will then merge into master soon. 

> my suggested change regarding variable naming
 
Unlesss I'm missing something, I don't see comment in regard to variable naming.

**Comment by emiliom** [comment]
@lsetiawan I'll hold off on approving and merging, until I hear back from you about my broader comment plus my suggestion to rename a variable. But nice job!

Also, let's make sure Jeff et al are very aware of this change. Correct me if I'm wrong, but all existing code that uses output from this function will need to be upgraded to the case-sensitive dataframe column names, right?

File: odm2api/ODM2/services/readService.py (hunk 2 above)
Comment (lsetiawan) [reply odm2api/ODM2/services/readService.py]: I needed to refresh to see this comment, I'll make that change. Thanks.

**Comment by emiliom** [comment]
> This is a PR to the dev branch that we have been merging into. This dev branch will then merge into master soon.

Great. My bad! I thought I checked for that, but obviously I didn't! My excuse is that this is my first PR review in a month, and ... also ... this year :stuck_out_tongue_winking_eye:

File: odm2api/ODM2/services/readService.py (hunk 2 above)
Comment (emiliom) [reply odm2api/ODM2/services/readService.py]: That was fast. Great.

**Comment by emiliom** [comment]
Ok, I'll merge once the CI's are done.

File: odm2api/ODM2/services/readService.py (hunk 1 above)
Comment (lsetiawan) [reply odm2api/ODM2/services/readService.py]: Thanks :smile:

**Comment by lsetiawan** [comment]
> Correct me if I'm wrong, but all existing code that uses output from this function will need to be upgraded to the case-sensitive dataframe column names, right?

Yes, if a code is doing `df['datavalue'] ` for example, it won't work anymore.

**Comment by emiliom** [comment]
I'm assuming we can ignore the AppVeyor failures. So, going ahead and merging the PR.

**Comment by lsetiawan** [comment]
Awesome! Thanks.

**Pull request merged**